### PR TITLE
Inventory Strategy lifo-fifo enumerations

### PR DESCRIPTION
**Pull request by davobag**
#Inventory Strategy lifo-fifo enumerations

### CODE REVIEW

File: sources/DavoodBaghdadi/InventoryStrategy/InventoryWrapper.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        Products.Add(product);
+    }
+
+    public IEnumerator<Product> GetEnumerator()
+    {
+        switch (_managementApproach)
+        {
+            case InventoryManagementApproach.LastInFirstOut:
+                return new LIFOEnumerator(Products);
+            default: return new FIFOEnumerator(Products);
```
Comment (thisisnabi) [inline sources/DavoodBaghdadi/InventoryStrategy/InventoryWrapper.cs:29]: You must enter the cases specifically.

Change it into the 

```

case InventoryManagementApproach.LastInFirstOut:
default:
        return....

```

File: sources/DavoodBaghdadi/InventoryStrategy/Enumerators/FIFOEnumerator.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    public bool MoveNext()
+    {
+        if (_currentPosition < _products.Count)
+        {
+            _currentPosition++;
+            return true;
+        }
+        else { return false; }
```
Comment (thisisnabi) [inline sources/DavoodBaghdadi/InventoryStrategy/Enumerators/FIFOEnumerator.cs:47]: You can remove the else statement and just return `false`

**Review by thisisnabi — CHANGES_REQUESTED** [review]
Please solve these issues.

**Comment by davobag** [comment]
Thanks dear Nabi !
I resolved your comments in local repository, so I did not push according to merge and close the request.

**Pull request merged**